data-raw/qc-markdowns/adult-holding-redd-and-carcass-surveys/feather-river/feather-carcass-survey-qc.Rmd

```diff
@@ -0,0 +1,539 @@
+---
+title: "Feather Carcass QC "
+author: "Inigo Peng"
+date: '2022-08-18'
+output: rmarkdown::github_document
+---
+  
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, warning = FALSE, fig.width=15, fig.height=10)
+
+library(tidyverse)
+library(lubridate)
+library(googleCloudStorageR)
+library(ggplot2)
+library(scales)
+library(readxl)
+library(janitor)
+library(hms) #?as_hms()
+library(RODBC)
+library(knitr)
+library(wesanderson)
+```
+
+
+# Feather River Carcass Data
+
+## Description of Monitoring Data
+
+**Timeframe:** 2000 to 2016
+  
+**Video Season:** 
+  
+**Completeness of Record throughout timeframe:** 
+
+* Data collected varies between different seasons. Cwt has the highest variability. Stop collecting "Chops Recovery" after (2011?); earlier years have more detailed chops data and had separate table for tags data; later chops data and tags data combines into one.
+  
+**Sampling Location:** Various sampling locations on Feather River
+  
+**Data Contact:** 
+  
+Any additional info?
+
+## Access Cloud Data
+
+```{r, eval=FALSE}
+# Run Sys.setenv() to specify GCS_AUTH_FILE and GCS_DEFAULT_BUCKET before running 
+# getwd() to see how to specify paths 
+# Open object from google cloud storage
+# Set your authentication using gcs_auth
+gcs_auth(json_file = Sys.getenv("GCS_AUTH_FILE"))
+# Set global bucket 
+gcs_global_bucket(bucket = Sys.getenv("GCS_DEFAULT_BUCKET"))
+
+# git data and save as xlsx
+```
+
+
+```{r}
+read_from_cloud <- function(carcass_data_type, year){
+  gcs_get_object(object_name = paste0("adult-holding-redd-and-carcass-surveys/feather-river/data/feather_carcass_",carcass_data_type, "_",year, ".csv"),
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = paste0("feather_carcass_",carcass_data_type,"_",year, ".csv"),
+               overwrite = TRUE)
+  data <- read.csv(paste0("feather_carcass_",carcass_data_type, "_", year, ".csv"))  
+}
+```
+
+### 2000 Chops and Tags
+*2000 and 2001 chops and tags are in different tables
+*2000 grilse has sex information
+
+```{r}
+feather_carcass_chops_2000 <- read_from_cloud("chops", 2000) %>% 
+  rename(adult_male = "male_chop",
+         adult_female = "female_chop",
+         sec = "sect") %>% 
+  pivot_longer(cols = c("adult_male", "adult_female", "grilse_male",
+                        "grilse_female"),
+               names_to = c("lifestage", "sex"),
+               names_sep ="_",
+               values_to = "chop_count") %>% 
+  mutate(chop_clip_status = "unknown",
+           tag_clip_status = "unknown")
+
+feather_carcass_tags_2000 <- read_from_cloud("tags", 2000) %>% 
+  mutate(tag_count = 1)
+
```

Comment:
Was this resolved? I remember we went back together and looked at this. Lets remove comment if resolved.